### ISSUE TITLE
ds.join better control for wait=True

### ISSUE THREAD

**Issue by jamesrkg** [Crunch-io/scrunch#416]
When using `wait=True` we get only the pycrunch's default 30 second timeout window, which is very frequently not long enough and leads to numerous timeouts that can't be controlled for using the current join method.

`timeout=30` should be added to the kwargs for `join` so that the length of the timeout window can be controlled, and the `wait` section in the method should change to:

```python
from pycrunch.progress import DefaultProgressTracking

def join(left_ds, left_var, right_ds, right_var, columns=None, filter=None, wait=True, timeout=30):

    ...

    if wait:
        progress_tracking = DefaultProgressTracking(timeout, interval=0.5)
        return wait_progress(
            r=progress,
            session=left_ds.resource.session,
            entity=left_ds,
            progress_tracker=progress_tracking
        )
```

**Comment by jjdelc** [comment]
We have a PR that removes the `wait` argument and brings a `timeout` argument to match other calls. https://github.com/Crunch-io/scrunch/pull/421

The `wait` argument seemed redundant as the `.join()` response would not be very useful if it was ever sent as `False`. So the default behavior will be always blocking with a default timeout that can be changed by the caller.

**Comment by jamesrkg** [comment]
Fine for me, thanks @jjdelc.

**Comment by jjdelc** [comment]
Released in v0.10.0
